src/pe/optional_header.rs

```diff
@@ -234,6 +235,40 @@ impl From<WindowsFields32> for WindowsFields {
 
 pub type WindowsFields = WindowsFields64;
 
+impl WindowsFields {
+    /// Validate the Windows-Specific Fields
+    ///
+    /// This method DOES NOT validate `check_sum` field since
+    /// the algorithm for computing the checksum is incorporated
+    /// into IMAGEHELP.DLL, which is os-dependent.
+    pub fn validate(&self) -> error::Result<()> {
+        let mut error_messages = vec![];
+        if self.image_base % 0x1000 != 0 {
+            error_messages.push("ImageBase must be a multiple of 64K.".to_owned());
+        }
+        if self.section_alignment < self.file_alignment {
+            error_messages.push(
+                "SectionAlignment must be greater than or equal to FileAlignment.".to_owned(),
+            );
+        }
+        if self.file_alignment < 512 || self.file_alignment > 0x1000 || self.file_alignment % 2 != 0
+        {
+            error_messages
+                .push("FileAlignment should be a power of 2 between 512 and 64K.".to_owned());
+        }
+        if self.size_of_image % self.section_alignment != 0 {
+            error_messages.push("SizeOfImage must be a multiple of SectionAlignment.".to_owned());
+        }
+        if error_messages.is_empty() {
+            Ok(())
+        } else {
+            Err(error::Error::Malformed(
+                super::utils::organize_validation_error_messages("", error_messages),
```

Comment:
nit: This was bit confusing, seeing the `""` in most calls, but i see now it's a prefix. Not sure if `Option<&str>` is better? Also since it's `pub(crate)` doesn't matter that much I suppose, since it's more of an internal function to PE. Perhaps `pub(super)` is even better, I can't remember if this would work w.r.t. privacy requirements though?